### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by AnirudhSubramanyam**
This PR fixes several bugs. At this point, I think we can consider merging into master and tagging a new release.

### CODE REVIEW

File: src/Evaluators/ProxALEvalutor.jl [hunk 1]
```diff
@@ -268,9 +266,10 @@ function optimize!(nlp::ProxALEvaluator; print_timings=false)
         elapsed_t = @elapsed begin
             if algparams.updateτ && runinfo.iter > 1
                 maxθ = algparams.decompCtgs  ? max(algparams.θ_t, algparams.θ_c) : algparams.θ_t
-                delta = runinfo.lyapunov[end-1] - runinfo.lyapunov[end]
-                if delta < 0.0 && algparams.τ < 320.0*maxθ
-                    algparams.τ *= 1.5
+                delta = (runinfo.lyapunov[end-1] - runinfo.lyapunov[end])/abs(runinfo.lyapunov[end])
+                if delta < -1e-4 && algparams.τ < 320.0*maxθ
+                    # @show(delta)
```
Comment (frapac) [inline src/Evaluators/ProxALEvalutor.jl:271]: I would maybe remove dead comment?

File: src/OPF/opfmodel.jl [hunk 2]
```diff
@@ -344,6 +344,8 @@ function compute_time_linking_constraints(
     if modelinfo.time_link_constr_type == :inequality
         link[:ramping_p] = [(t > 1) ? (+Pg[g,1,t-1] - Pg[g,1,t] - gens[g].ramp_agc) : 0.0 for g=1:ngen,t=Trange]
         link[:ramping_n] = [(t > 1) ? (-Pg[g,1,t-1] + Pg[g,1,t] - gens[g].ramp_agc) : 0.0 for g=1:ngen,t=Trange]
+    elseif modelinfo.time_link_constr_type == :equality
+        link[:ramping] = [(t > 1) ? (Pg[g,1,t-1] - Pg[g,1,t] + St[g,t] - gens[g].ramp_agc) : 0.0 for g=1:ngen,t=Trange]
```
Comment (frapac) [inline src/OPF/opfmodel.jl:348]: here, `t` is directly set equal to the horizon `t=Trange`. Is it expected? Or do we want `t` to vary (`t=1:Trange`)?

File: src/OPF/opfmodel.jl [hunk 3]
```diff
@@ -372,10 +374,16 @@ function compute_ctgs_linking_constraints(
     if modelinfo.ctgs_link_constr_type == :corrective_inequality
         link[:ctgs_p] = [(k > 1) ? (+Pg[g,1,t] - Pg[g,k,t] - gens[g].scen_agc) : 0.0 for g=1:ngen,k=Krange,t=Trange]
         link[:ctgs_n] = [(k > 1) ? (-Pg[g,1,t] + Pg[g,k,t] - gens[g].scen_agc) : 0.0 for g=1:ngen,k=Krange,t=Trange]
+    elseif modelinfo.ctgs_link_constr_type == :corrective_equality
+        link[:ctgs] = [(k > 1) ? (Pg[g,1,t] - Pg[g,k,t] + Sk[g,k,t] - gens[g].scen_agc) : 0.0 for g=1:ngen,k=Krange,t=Trange]
```
Comment (frapac) [inline src/OPF/opfmodel.jl:378]: Same here, both `t` and `k` are fixed

File: src/OPF/opfmodel.jl [hunk 4]
```diff
@@ -499,25 +507,25 @@ function compute_proximal_function(
...
     t = block[2]
 
-    prox_pg = sum((x1.Pg[:,k,t] .- x2.Pg[:,k,t]).^2)
+    prox_pg = algparams.τ*sum((x1.Pg[:,k,t] .- x2.Pg[:,k,t]).^2)
     prox_penalty = 0
 
     if modelinfo.time_link_constr_type == :penalty
         if t > 1 && k == 1
-            prox_penalty += sum((x1.Zt[:,t] .- x2.Zt[:,t]).^2)
+            prox_penalty += (algparams.ρ_t/32.0)*sum((x1.Zt[:,t] .- x2.Zt[:,t]).^2)
```
Comment (frapac) [inline src/OPF/opfmodel.jl:515]: Here, I would maybe add a short comment to explain why we are dividing `rho` by `32.0`

**Review by frapac — COMMENTED** [review]
The PR looks overall good to me! I just left a few comments, about some points that were unclear to me in the code. 
I agree that once this PR merged, we could do a release on master.

File: src/OPF/opfmodel.jl (hunk 2 above)
Comment (AnirudhSubramanyam) [reply src/OPF/opfmodel.jl:348]: Note that `Trange` is already defined as a range: `Trange = (tIdx == 0) ? (1:T) : (tIdx:tIdx)`. The case `tIdx:tIdx` is used in the distributed case.

File: src/Evaluators/ProxALEvalutor.jl (hunk 1 above)
Comment (AnirudhSubramanyam) [reply src/Evaluators/ProxALEvalutor.jl:271]: done! Thanks!

File: src/OPF/opfmodel.jl (hunk 4 above)
Comment (AnirudhSubramanyam) [reply src/OPF/opfmodel.jl:515]: done! Thanks!

File: src/OPF/opfmodel.jl (hunk 2 above)
Comment (frapac) [reply src/OPF/opfmodel.jl:348]: Oh, I see! Sorry, I didn't understand the true nature of `Trange`...

**Pull request merged**